src/libraries/System.Runtime/tests/System/ByteTests.GenericMath.cs

```diff
@@ -0,0 +1,1175 @@
+        {
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Max((byte)0x00, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Max((byte)0x01, (byte)1));
+            Assert.Equal((byte)0x7F, NumberHelper<byte>.Max((byte)0x7F, (byte)1));
+            Assert.Equal((byte)0x80, NumberHelper<byte>.Max((byte)0x80, (byte)1));
+            Assert.Equal((byte)0xFF, NumberHelper<byte>.Max((byte)0xFF, (byte)1));
+        }
+
+        [Fact]
+        public static void MinTest()
+        {
+            Assert.Equal((byte)0x00, NumberHelper<byte>.Min((byte)0x00, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Min((byte)0x01, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Min((byte)0x7F, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Min((byte)0x80, (byte)1));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Min((byte)0xFF, (byte)1));
+        }
+
+        [Fact]
+        public static void SignTest()
+        {
+            Assert.Equal((byte)0x00, NumberHelper<byte>.Sign((byte)0x00));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Sign((byte)0x01));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Sign((byte)0x7F));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Sign((byte)0x80));
+            Assert.Equal((byte)0x01, NumberHelper<byte>.Sign((byte)0xFF));
+        }
+
+        [Fact]
+        public static void TryCreateFromByteTest()
+        {
+            byte result;
+
+            Assert.True(NumberHelper<byte>.TryCreate<byte>(0x00, out result));
+            Assert.Equal((byte)0x00, result);
+
+            Assert.True(NumberHelper<byte>.TryCreate<byte>(0x01, out result));
+            Assert.Equal((byte)0x01, result);
+
+            Assert.True(NumberHelper<byte>.TryCreate<byte>(0x7F, out result));
+            Assert.Equal((byte)0x7F, result);
+
+            Assert.True(NumberHelper<byte>.TryCreate<byte>(0x80, out result));
+            Assert.Equal((byte)0x80, result);
+
+            Assert.True(NumberHelper<byte>.TryCreate<byte>(0xFF, out result));
+            Assert.Equal((byte)0xFF, result);
+        }
+
+        [Fact]
+        public static void TryCreateFromCharTest()
+        {
+            byte result;
+
+            Assert.True(NumberHelper<byte>.TryCreate<char>((char)0x0000, out result));
+            Assert.Equal((byte)0x00, result);
+
+            Assert.True(NumberHelper<byte>.TryCreate<char>((char)0x0001, out result));
+            Assert.Equal((byte)0x01, result);
+
+            Assert.False(NumberHelper<byte>.TryCreate<char>((char)0x7FFF, out result));
+            Assert.Equal((byte)0x00, result);
+
+            Assert.False(NumberHelper<byte>.TryCreate<char>((char)0x8000, out result));
+            Assert.Equal((byte)0x00, result);
+
+            Assert.False(NumberHelper<byte>.TryCreate<char>((char)0xFFFF, out result));
+            Assert.Equal((byte)0x00, result);
+        }
```

Comment:
Did you consider using theories to tighten up these tests and avoid the duplication? e.g. I prefer the conciseness and clarity of this:
```suggestion
        [Theory]
        [InlineData(0x0000, 0x00)]
        [InlineData(0x0001, 0x01)]
        [InlineData(0x7FFF, 0x00)]
        [InlineData(0x8000, 0x00)]
        [InlineData(0xFFFF, 0x00)]
        public static void TryCreateFromCharTest(int input, int expected)
        {
            Assert.True(NumberHelper<byte>.TryCreate<char>((char)input, out byte result));
            Assert.Equal((byte)expected, result);
        }
```

Reply:
I'll log an issue for this as a low hanging fruit to make these easier to manage and add new cases for in the future.